providers/stores/abstractFileStore.js

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const utils = require('../../lib/utils')
+const fs = require('fs')
+const path = require('path')
+const recursive = require('recursive-readdir')
+const { promisify } = require('util')
+const ResultCoordinates = require('../../lib/resultCoordinates')
+
+class AbstractFileStore {
+  constructor(options) {
+    this.options = options
+  }
+
+  async initialize() {}
+
+  /**
+   * Visit all of the files associated with the given coordinates.
+   *
+   * @param {EntityCoordinates} coordinates - Accepts partial coordinates.
+   * @returns The collection of results returned by the visitor
+   */
+  async list(coordinates, visitor) {
+    try {
+      const paths = await recursive(this._toStoragePathFromCoordinates(coordinates), ['.DS_Store'])
+      return await Promise.all(
+        paths.map(async path => {
+          if (!this._isValidPath(path)) return null
+          const data = await promisify(fs.readFile)(path)
+          return data ? visitor(JSON.parse(data)) : null
+        })
```

Comment:
should we add `.filter(x => x)` after this `.map()` since we are returning null in some situations